src/inet/TCPEndPointImplLwIP.cpp

```diff
@@ -512,7 +512,7 @@ CHIP_ERROR TCPEndPointImplLwIP::AckReceive(uint16_t len)
     LOCK_TCPIP_CORE();
 
     if (mTCP != nullptr)
-        tcp_recved(mTCP, len);
+        tcp_recved(mTCP, static_cast<uint16_t>(len));
```

Comment:
Why a uint16_t here?

Reply:
Yes, all LwIP APIs are limited to u16 which makes it problematic to deal with. So, for LwIP based systems, one solution would be to only accept TCP payloads that are less than UINT16_MAX, else return an appropriate error.